books/kestrel/utilities/system/numbered-names.lisp

```diff
@@ -626,17 +662,16 @@
               (posp val)))
 
 (define get-global-numbered-name-index ((wrld plist-worldp))
-  :returns (global-index "A @(tsee posp).")
-  :verify-guards nil
+  :returns (global-index posp "A @(tsee posp).")
   :parents (global-numbered-name-index)
   :short "Retrieve the global index for numbered names."
   :long
   "<p>
    If the global index is not set yet, 1 is returned.
    </p>"
   (let ((pair (assoc-eq 'index
-                        (table-alist 'global-numbered-name-index wrld))))
-    (if pair (cdr pair) 1)))
+                        (table-alist+ 'global-numbered-name-index wrld))))
+    (pos-fix (cdr pair))))
```

Comment:
You may also consider throwing an internal error if the expected `posp` condition is violated.